infrastructure/instance_metadata_settings_source.go

```diff
@@ -0,0 +1,66 @@
+package infrastructure
+
+import (
+	"encoding/json"
+	"fmt"
+
+	boshplatform "github.com/cloudfoundry/bosh-agent/platform"
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	bosherr "github.com/cloudfoundry/bosh-utils/errors"
+	boshlog "github.com/cloudfoundry/bosh-utils/logger"
+)
+
+type InstanceMetadataSettingsSource struct {
+	metadataHost    string
+	metadataHeaders map[string]string
+	settingsPath    string
+
+	platform boshplatform.Platform
+	logger   boshlog.Logger
+
+	logTag          string
+	metadataService DynamicMetadataService
+}
+
+func NewInstanceMetadataSettingsSource(
```

Comment:
i think we should add new _test.go for this class based on https://github.com/cloudfoundry/bosh-agent/blob/master/infrastructure/http_metadata_service_test.go.